olive/engine/engine.py

```diff
@@ -323,7 +323,15 @@ def run(
 
         outputs = {}
         pf_footprints = {}
+
+        origin_target_system = self.target
+        origin_host_system = self.host
         for accelerator_spec in self.accelerator_specs:
+            if origin_target_system.olive_managed_env is True:
+                self.target = create_new_system(origin_target_system, accelerator_spec)
```

Comment:
could we create the system only when needed? for example, in AML scenarios, the cost to create different AML system might be time consuming.

Reply:
This system is only created when user wants Olive to manage environment and these created systems will be used for pass runing and evaluation.